src/components/prompt-form.tsx

```diff
@@ -0,0 +1,130 @@
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Placeholder } from "@tiptap/extension-placeholder";
+import { EditorProvider } from "@tiptap/react";
+import { StarterKit } from "@tiptap/starter-kit";
+import { ArrowUp } from "lucide-react";
+import { useRouter } from "next/navigation";
+import { useRef } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { v4 } from "uuid";
+import { z } from "zod";
+
+import { useChats } from "@/hooks/use-chats";
+import { faculties } from "@/lib/faculties";
+import type { Chat } from "@/types/chat";
+
+import { Button } from "./ui/button";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "./ui/select";
+
+const extensions = [
+  StarterKit,
+  Placeholder.configure({
+    placeholder: "Wpisz temat lub opis swojej pracy...",
+  }),
+];
+
+const formSchema = z.object({
+  prompt: z.string().min(1),
+  faculty: z.string().default("any"),
+});
+
+export function PromptForm() {
+  const router = useRouter();
+  const { addChat } = useChats();
+  const formRef = useRef<null | HTMLFormElement>(null);
+
+  const {
+    handleSubmit,
+    setValue,
+    getValues,
+    control,
+    formState: { isSubmitting },
+  } = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+  });
+
+  const onSubmit = () => {
+    const { prompt, faculty } = getValues();
```

Comment:
Should be fixed in https://github.com/Solvro/web-promochator/pull/58/commits/b6478597a70ea190e298710eef904f1357b2e65b but please double check on your end. In my previous implementation I misunderstood and misused zod's `.default()` method, the proper way is to set `defaultValue` prop in react hook form's `<Controller/>` component. Shadcn's select component does not allow an item to have an empty string as its value because its reserved for the placeholder, so instead the default value is "any" with "Dowolny wydział". Then the `onSubmit` function checks if `faculty` value is "any" - if so, we send an empty string in the request, otherwise the faculty name